### ISSUE TITLE
[2.0] "An async callback was invoked more than once, which could be a sign of a defect" logged by ZIO runtime.

### ISSUE THREAD

**Issue by artur-jablonski** [zio/zio#7163]
After running this ZStream for a while (few seconds):

```scala
    ZStream
      .fromIterable(List(1, 2, 3, 4, 5))
      .flatMapPar(1) { i =>
        ZStream.fail("kaboom!")
      }
      .debug("debug")
      .retry(Schedule.fixed(1.second))
      .runDrain
```

ZIO runtime logs out "An async callback was invoked more than once, which could be a sign of a defect"

Here's example of output from running the program:

```
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
timestamp=2022-08-02T13:13:29.161347842+02 level=DEBUG thread=zio-fiber-54 message="An async callback was invoked more than once, which could be a sign of a defect: Failure(Interrupt(Runtime(52,1659438809155,MainExploring.run(MainExploring.scala:25)),Stack trace for thread "zio-fiber-":
))" cause=
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
timestamp=2022-08-02T13:13:30.157272451+02 level=DEBUG thread=zio-fiber-58 message="An async callback was invoked more than once, which could be a sign of a defect: Failure(Interrupt(Runtime(57,1659438810154,MainExploring.run(MainExploring.scala:25)),Stack trace for thread "zio-fiber-":
))" cause=
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)
<FAIL> debug: Fail(kaboom!,Stack trace for thread "zio-fiber-":
)

```

`MainExploring.sacala:25` points to `.flatMapPar(1) { i =>` in the source code. 

Not sure if this is an issue, everything seems to be working fine.

**Comment by adamgfraser** [comment]
@artur-jablonski Hi! This was previously resolved by #7083 so going to close this.

**Comment by artur-jablonski** [comment]
Fair play @adamgfraser! Thanks!
